src/Acme.UI/Infrastructure/Handlers/MemberSaveHandler.cs

```diff
@@ -0,0 +1,147 @@
+                        {
+                            var node = _contentService.Create(member.Name, Guid.Parse(parentNodeIdAndKey.Item2), "physiotherapist");
+                            node.SetValue("physio_name", member.Name);
+                            node.SetValue("physio_email", member.Email);
+
+                            var address = GetUmbracoPropertyValue(member.Properties, addressAlias);
+                            var phone = GetUmbracoPropertyValue(member.Properties, phoneAlias);
+                            node.SetValue("physio_address", address);
+                            node.SetValue("physio_contact", phone);
+
+                            _contentService.SaveAndPublish(node);
+                            processedNodesIds.Add(member.Id.ToString());
+
+                            member.SetValue(memberDataFolderAlias, node.GetUdi());
+                        }
+                    }
+                }
+            }
+        }
+ 
+        private bool MandatoryFieldsArePopulated(IMember member)
+        {
+            if (string.IsNullOrWhiteSpace(member.Name) || string.IsNullOrWhiteSpace(member.Email))
+            {
+                return false;
+            }
+
+            var address = GetUmbracoPropertyValue(member.Properties, addressAlias);
+            var phone = GetUmbracoPropertyValue(member.Properties, phoneAlias);
+
+            if (string.IsNullOrWhiteSpace(address) || string.IsNullOrWhiteSpace(phone))
+            {
+                return false;
+            }
+
+            return true;
+        } 
+
+        private Tuple<string, string> GetParentNodeIdAndKey()
+        {
+            if (ExamineManager.Instance.TryGetIndex("ExternalIndex", out var index))
+            {
+                var searcher = index.GetSearcher();
+                var results = searcher.CreateQuery("content").NodeTypeAlias("physiotherapists").Execute();
+
+                if (results.Any())
+                {
+                    var node = results.First();
+                    var keyValue = node.Values.FirstOrDefault(x => "__Key".Equals(x.Key));
+                    var key = keyValue.Value;
+                    return new Tuple<string, string>(node.Id, key);
+                }
+            }
+
+            return new Tuple<string, string>(null, null);
+        }
+
+        private bool nodeExists(string parentNodeId, string nodeName)
+        {
+            if (ExamineManager.Instance.TryGetIndex("ExternalIndex", out var index))
+            {
+                var searcher = index.GetSearcher();
+                var results = searcher.CreateQuery("content").ParentId(int.Parse(parentNodeId)).And().NodeName(nodeName).Execute();
+
+                if (results.Any())
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private string GetUmbracoPropertyValue(PropertyCollection properties, string propertyAlias)
```

Comment:
Pull this into a helper class you will likely use this many times more and not isolated to MemberSave Handler

Reply:
please have  a look at this one

Reply:
In my opinion the common part is from "ExamineManager.Instance....." to "CreateQuery". The clauses can change and can be nested with logical operators. So I am making only the common portion part of the helper class. Will be happy to extend the functionality further if you think otherwise.

Reply:
n my opinion the common part is from "ExamineManager.Instance....." to "CreateQuery". The clauses can change and can be nested with logical operators. So I am making only the common portion part of the helper class. Will be happy to extend the functionality further if you think otherwise.